src/policy/immix/immixspace.rs

```diff
@@ -646,14 +721,25 @@ pub struct PrepareBlockState<VM: VMBinding> {
 
 impl<VM: VMBinding> PrepareBlockState<VM> {
     /// Clear object mark table
-    fn reset_object_mark(chunk: Chunk) {
+    fn reset_object_mark(&self) {
         // NOTE: We reset the mark bits because cyclic mark bit is currently not supported, yet.
         // See `ImmixSpace::prepare`.
         if let MetadataSpec::OnSide(side) = *VM::VMObjectModel::LOCAL_MARK_BIT_SPEC {
-            side.bzero_metadata(chunk.start(), Chunk::BYTES);
+            side.bzero_metadata(self.chunk.start(), Chunk::BYTES);
+        }
+        if self.space.space_args.reset_log_bit_in_major_gc {
+            if let MetadataSpec::OnSide(side) = *VM::VMObjectModel::GLOBAL_LOG_BIT_SPEC {
+                // We zero all the log bits in major GC, and for every object we trace, we will mark the log bit again.
+                side.bzero_metadata(self.chunk.start(), Chunk::BYTES);
+            } else {
+                // If the log bit is not in side metadata, we cannot bulk zero. We can either
+                // clear the bit for dead objects in major GC, or clear the log bit for new
+                // objects. In both cases, we do not need to set log bit at tracing.
```

Comment:
In English, it is more natural to use "in either case" if the main part is negative, i.e. "we do *not*".

```suggestion
                // objects. In either case, we do not need to set log bit at tracing.
```

See: https://forum.wordreference.com/threads/in-either-case-vs-in-both-case.2243816/